web/src/engine/websites/Sukima.ts

```diff
@@ -0,0 +1,191 @@
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('sukima', `Sukima`, 'https://www.sukima.me', Tags.Language.Japanese, Tags.Media.Manga, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return /https?:\/\/www\.sukima\.me\/book\/title\/\S+\/$/.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const id = new URL(url).pathname.match(/\/(\w+)\/?$/)[1];
+        const request = new FetchRequest(new URL(`/api/book/v1/title/${id}/`, this.URI).href);
+        const data = await FetchJSON<APIManga>(request);
+        return new Manga(this, provider, id, data.title);
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList = [];
+        for (let page = 1, run = true; run; page++) {
+            const mangas = await this.getMangasFromPage(provider, page);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        const categorys = await this.fetchPOST<APICategories>('/api/book/v1/free/', JSON.stringify({store: false, genre: '0'}));
+        for (const category of categorys.rows) {
+            const uri = new URL(category.more_btn.link, this.URI);
+            if (uri.searchParams.get('tag') != null) {
+                for (let page = 1, run = true; run; page++) {
+                    const mangas = await this.getMangasFromPage(provider, page, [uri.searchParams.get('tag')]);
+                    mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+                }
+            }
+        }
+        return mangaList;
+    }
+
+    async getMangasFromPage(provider: MangaPlugin, page: number, tag :string[]= []): Promise<Manga[]>{
+        const body = {
+            'page': page,
+            'sort_by': 0,
+            'tag': tag
+        };
+        const pageContent = await this.fetchPOST<APIMangas>('/api/v1/search/', JSON.stringify(body));
+        const mangas = pageContent.items.map(element => new Manga(this, provider, element.title_code, element.title_name));
+        return page > pageContent.max_page ? [] : mangas;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const request = new FetchRequest(new URL(`/api/book/v1/title/${manga.Identifier}/`, this.URI).href);
+        const books = await FetchJSON<APIManga>(request);
+        const chapters = [];
+        for (const book of books.contents) {
+            for (const chapter of book.stories) {
+                const id = `/bv/t/${chapter.title_code}/v/${chapter.volume}/s/${chapter.story}/p/1`;
+                const title = `(${chapter.volume.toString().padStart(3, '0')}-${chapter.story.toString().padStart(3, '0')})${chapter.info.text.trim()}`.replace(manga.Title, '').trim();
```

Comment:
If there are chapters with more then 999 chapters, the padding of 3 digits would be insufficient